sdk/core/Azure.Core/src/Shared/Utf8JsonReaderExtensions.cs

```diff
@@ -0,0 +1,166 @@
+        {
+            { typeof(int), str => Convert.ToInt32(str) },
+            { typeof(string), str => str },
+        };
+
+        private static string ConvertKeyToString(Type typeToConvert, string key) => key;
+
+        public static List<T> GetList<T>(this ref Utf8JsonReader reader, ModelSerializerOptions options)
+        {
+            List<T> list = new List<T>();
+
+            if (reader.TokenType == JsonTokenType.Null)
+                return null;
+
+            if (reader.TokenType != JsonTokenType.StartArray)
+                throw new InvalidOperationException("Expected StartArray token");
+
+            while (reader.Read())
+            {
+                if (reader.TokenType == JsonTokenType.EndArray)
+                    break;
+
+                list.Add(reader.GetObject<T>(options));
+            }
+
+            return list;
+        }
+
+        public static Dictionary<TKey, TValue> GetDictionary<TKey, TValue>(this ref Utf8JsonReader reader, ModelSerializerOptions options)
+            where TKey : notnull
+        {
+            if (!_keyConverter.TryGetValue(typeof(TKey), out var converter))
+                throw new InvalidOperationException($"Cannot use type {typeof(TKey).Name} as a key for a dictionary");
+
+            Dictionary<TKey, TValue> dictionary = new Dictionary<TKey, TValue>();
+            reader.Read();
+
+            if (reader.TokenType == JsonTokenType.Null)
+                return null;
+
+            if (reader.TokenType != JsonTokenType.StartObject)
+                throw new InvalidOperationException("Expected StartObject token");
+
+            while (reader.Read())
+            {
+                if (reader.TokenType == JsonTokenType.EndObject)
+                    break;
+
+                if (reader.TokenType != JsonTokenType.PropertyName)
+                    throw new InvalidOperationException("Expected PropertyName token");
+
+                var propertyName = reader.GetString();
+                reader.Read();
+                dictionary.Add((TKey)converter(propertyName), reader.GetObject<TValue>(options));
+            }
+
+            return dictionary;
+        }
+
+        public static T GetObject<T>(this ref Utf8JsonReader reader, ModelSerializerOptions options)
+        {
+            var typeToConvert = typeof(T);
+
+            if (_readerActions.TryGetValue(typeToConvert, out var readValue))
+            {
+                return (T)readValue(ref reader);
+            }
+
+            var model = Activator.CreateInstance(typeToConvert, true) as IModelJsonSerializable<T>;
+            if (model is null)
+                throw new InvalidOperationException($"{typeToConvert.Name} does not implement {nameof(IModelJsonSerializable<T>)}");
+
+            return (T)model.Deserialize(ref reader, options);
+        }
+
+        public static bool TryDeserialize<T>(this ref Utf8JsonReader reader, ModelSerializerOptions options, PropertySetter<T> setProperty, out T properties)
```

Comment:
Where this method is going to be used?

Reply:
https://github.com/Azure/azure-sdk-for-net/pull/37970